pyls_black/plugin.py

```diff
@@ -50,29 +50,22 @@ def format_document(document, range=None):
 
 
 def format_text(*, text, config):
-    line_length = config["line_length"]
-    fast = config["fast"]
-    mode = black.FileMode.from_configuration(
-        py36=config["py36"],
-        pyi=config["pyi"],
-        skip_string_normalization=config["skip_string_normalization"],
-        skip_numeric_underscore_normalization=config[
-            "skip_numeric_underscore_normalization"
-        ],
-    )
-    return black.format_file_contents(
-        text, line_length=line_length, fast=fast, mode=mode
+    mode = black.FileMode(
+        target_versions=config["target_version"],
```

Comment:
This fails whenever working on a project that doesn't have a ``pyproject.toml`` file. Is that expected?

```
Traceback (most recent call last):
  File "/Users/fsouza/.venvs/vim/lib/python3.7/site-packages/pyls_jsonrpc/endpoint.py", line 113, in consume
    self._handle_request(message['id'], message['method'], message.get('params'))
  File "/Users/fsouza/.venvs/vim/lib/python3.7/site-packages/pyls_jsonrpc/endpoint.py", line 182, in _handle_request
    handler_result = handler(params)
  File "/Users/fsouza/.venvs/vim/lib/python3.7/site-packages/pyls_jsonrpc/dispatchers.py", line 23, in handler
    return method(**(params or {}))
  File "/Users/fsouza/.venvs/vim/lib/python3.7/site-packages/pyls/python_ls.py", line 274, in m_text_document__formatting
    return self.format_document(textDocument['uri'])
  File "/Users/fsouza/.venvs/vim/lib/python3.7/site-packages/pyls/python_ls.py", line 202, in format_document
    return self._hook('pyls_format_document', doc_uri)
  File "/Users/fsouza/.venvs/vim/lib/python3.7/site-packages/pyls/python_ls.py", line 118, in _hook
    return hook_handlers(config=self.config, workspace=self.workspace, document=doc, **kwargs)
  File "/Users/fsouza/.venvs/vim/lib/python3.7/site-packages/pluggy/hooks.py", line 284, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/Users/fsouza/.venvs/vim/lib/python3.7/site-packages/pluggy/manager.py", line 68, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/Users/fsouza/.venvs/vim/lib/python3.7/site-packages/pluggy/_tracing.py", line 80, in __call__
    return outcome.get_result()
  File "/Users/fsouza/.venvs/vim/lib/python3.7/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/Users/fsouza/.venvs/vim/lib/python3.7/site-packages/pluggy/callers.py", line 52, in from_call
    result = func()
  File "/Users/fsouza/.venvs/vim/lib/python3.7/site-packages/pluggy/_tracing.py", line 78, in <lambda>
    outcome = _Result.from_call(lambda: self.oldcall(hook, hook_impls, kwargs))
  File "/Users/fsouza/.venvs/vim/lib/python3.7/site-packages/pluggy/manager.py", line 62, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/Users/fsouza/.venvs/vim/lib/python3.7/site-packages/pluggy/callers.py", line 203, in _multicall
    gen.send(outcome)
  File "/Users/fsouza/.venvs/vim/lib/python3.7/site-packages/pyls_isort/plugin.py", line 33, in pyls_format_document
    results = outcome.get_result()
  File "/Users/fsouza/.venvs/vim/lib/python3.7/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/Users/fsouza/.venvs/vim/lib/python3.7/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/Users/fsouza/.venvs/vim/lib/python3.7/site-packages/pyls_black/plugin.py", line 10, in pyls_format_document
    return format_document(document)
  File "/Users/fsouza/.venvs/vim/lib/python3.7/site-packages/pyls_black/plugin.py", line 36, in format_document
    formatted_text = format_text(text=text, config=config)
  File "/Users/fsouza/.venvs/vim/lib/python3.7/site-packages/pyls_black/plugin.py", line 54, in format_text
    target_versions=config["target_version"],
KeyError: 'target_version'
```

Should ``defaults`` include ``"target_version": set()``?

Reply:
Thanks @fsouza, I believe there's only a test for an empty config file rather than no config file at all.

Reply:
> Should defaults include "target_version": set()? I can send a PR.

Yes, the default should be an empty set.

Reply:
Yeah writing that test is kinda tricky because black look for the config file recursively, so we'd need to potentially create a "project" in a temp directory or something like that? And technically that test could still fail if ran on a machine with a pyproject.toml under ``/tmp`` or ``/`` (dunno why someone would do that, but... lol)

Reply:
Sent the PR, without a test: https://github.com/rupert/pyls-black/pull/21

Can add the test if you think it makes sense (I'll be afk for the next 40-50 minutes though).